### PR TITLE
Xbrl context

### PR DESCRIPTION
**Pull request by mmoghaddam385**
For our purposes, XBRL consists of contexts, facts, and units. This is part one: the contexts

### CODE REVIEW

File: context.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+// Entity documents the business entity for a Context (business, government department, individual, etc.).
+// https://www.xbrl.org/Specification/XBRL-2.1/REC-2003-12-31/XBRL-2.1-REC-2003-12-31+corrected-errata-2013-02-20.html#_4.7.3
+type Entity struct {
+	Identifier Identifier `xml:"identifier"`
+	Segments   Segments   `xml:"segment"`
+}
+
+// Identifier specifies a scheme for identifying business entities and an identifier that follows the scheme.
+// https://www.xbrl.org/Specification/XBRL-2.1/REC-2003-12-31/XBRL-2.1-REC-2003-12-31+corrected-errata-2013-02-20.html#_4.7.3.1
+// For Example:
```
Comment (calebcase) [inline context.go:24]: <3 including examples of the data that might be parsed

File: context.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	// StartDate is non-nil and guaranteed to be before EndDate if Period.Type() returns Duration.
+	StartDate *string `xml:"startDate"`
+	// EndDate is non-nil and guaranteed to be after StartDate if Period.Type() returns Duration.
+	EndDate *string `xml:"endDate"`
+
+	// Instant is non-nil if Period.Type() returns Instant
+	Instant *string `xml:"instant"`
+
+	// Forever is non-nil if Period.Type() returns Forever.
+	// Note ideally this would be a bool, but the XML Unmarshaller doesn't support
```
Comment (calebcase) [inline context.go:85]: Should we make a custom unmarshal for it?

**Review by calebcase — APPROVED** [review]
Would there be any value in having a `(* Context) Validate()` that checked higher level semantics?

File: context.go (hunk 2 above)
Comment (mmoghaddam385) [reply context.go:85]: I don't think it's really necessary. `Type()` returning `Forever` is already a clean way to determine if the period represents 'forever', so you wouldn't really have to interact with this property directly as a user. 

I might come back to this later once I'm more experienced with custom unmarshallers though

**Comment by mmoghaddam385** [comment]
> Would there be any value in having a `(* Context) Validate()` that checked higher level semantics?

This might be valuable if/when we come back and add support for Context `Scenario`s, but for now it wouldn't really do much

**Pull request merged**